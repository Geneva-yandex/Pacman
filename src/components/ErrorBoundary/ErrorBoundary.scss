.ErrorBoundary {
  margin-top: 60px;
}
