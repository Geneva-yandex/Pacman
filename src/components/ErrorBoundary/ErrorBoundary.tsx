import {IErrorBoundaryState} from './types';
import * as React from 'react';
import './ErrorBoundary.scss';
import bem from 'easy-bem';

const b = bem('ErrorBoundary');
export default class ErrorBoundary extends React.Component<{}, IErrorBoundaryState> {
    constructor(props: {}) {
        super(props);
        this.state = {
            hasError: false
        };
    }

    static getDerivedStateFromError() {
        return {
            hasError: true
        };
    }

    componentDidCatch(error: Error, errorInfo: React.ErrorInfo) {
        console.error(error, errorInfo);
    }

    render() {
        if (this.state.hasError) {
            return (
                <div className={b()}>
                    <div className={'container-fluid'}>
                        <h1>Что-то пошло не так :(</h1>
                    </div>
                </div>
            );
        }

        return this.props.children;
    }
}
