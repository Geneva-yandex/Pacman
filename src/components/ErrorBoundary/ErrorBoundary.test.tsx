import * as React from 'react';
import {shallow} from 'enzyme';
import ErrorBoundary from './ErrorBoundary';

describe('ErrorBoundary', () => {
    test('ErrorBoundary component renders correctly', () => {
        const renderedComponent = shallow(<ErrorBoundary />);
        expect(renderedComponent).toMatchSnapshot();
    });
});

