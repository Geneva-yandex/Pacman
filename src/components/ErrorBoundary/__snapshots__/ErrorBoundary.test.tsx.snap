// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary ErrorBoundary component renders correctly 1`] = `ShallowWrapper {}`;
